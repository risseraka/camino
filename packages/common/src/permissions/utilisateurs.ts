import { isSuper, isAdministrationAdmin, isAdministrationEditeur, User } from '../roles'

export const canCreateEntreprise = (user: User): boolean => isSuper(user) || isAdministrationAdmin(user) || isAdministrationEditeur(user)
export const canCreateUtilisateur = (user: User): boolean => isSuper(user) || isAdministrationAdmin(user)
