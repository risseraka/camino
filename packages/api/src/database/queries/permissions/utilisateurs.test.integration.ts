import { IUtilisateur } from '../../../types'

import { dbManager } from '../../../../tests/db-manager'
import Utilisateurs from '../../models/utilisateurs'
import { utilisateursGet } from '../utilisateurs'
import { Administrations } from 'camino-common/src/static/administrations'
import options from '../_options'
import { testBlankUser, TestUser } from '../../../../tests/_utils'

console.info = jest.fn()
console.error = jest.fn()
beforeAll(async () => {
  await dbManager.populateDb()
  await Utilisateurs.query().insertGraph(mockUser, options.utilisateurs.update)
})

afterAll(async () => {
  await dbManager.closeKnex()
})

const mockAdministration = Administrations['aut-97300-01']

const mockUser: IUtilisateur = {
  id: 'utilisateurId',
  role: 'editeur',
  nom: 'utilisateurNom',
  email: 'utilisateurEmail',
  administrationId: mockAdministration.id,
  dateCreation: '2022-05-12'
}

describe('utilisateursQueryModify', () => {
  test.each<[TestUser, boolean]>([
    [{ role: 'super' }, true],
    [{ role: 'admin', administrationId: mockAdministration.id }, true],
    [{ role: 'editeur', administrationId: mockAdministration.id }, true],
    [{ role: 'lecteur', administrationId: mockAdministration.id }, true],
    [{ role: 'entreprise', entreprises: [] }, true],
    [{ role: 'defaut' }, false]
  ])(
    "Vérifie l'écriture de la requête sur un utilisateur",
    async (user, voit) => {
      const utilisateurs = await utilisateursGet(
        { noms: mockUser.nom },
        {},
        { ...user, ...testBlankUser }
      )
      if (voit) {
        expect(utilisateurs).toHaveLength(1)
        expect(utilisateurs[0]).toMatchSnapshot()
      } else {
        expect(utilisateurs).toHaveLength(0)
      }
    }
  )
})
