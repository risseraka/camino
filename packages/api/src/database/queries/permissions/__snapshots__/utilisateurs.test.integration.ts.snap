// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`utilisateursQueryModify Vérifie l'écriture de la requête sur un utilisateur 1`] = `
Object {
  "administration": Object {
    "activitesTypes": Array [],
    "id": "aut-97300-01",
    "titresTypes": Array [
      Object {
        "archive": null,
        "associee": true,
        "contenuIds": Array [
          Object {
            "elementId": "mecanise",
            "sectionId": "arm",
          },
          Object {
            "elementId": "franchissements",
            "sectionId": "arm",
          },
        ],
        "domaineId": "m",
        "gestionnaire": null,
        "id": "arm",
        "typeId": "ar",
      },
    ],
  },
  "administrationId": "aut-97300-01",
  "dateCreation": "2022-05-12",
  "email": "utilisateurEmail",
  "entreprises": Array [],
  "id": "utilisateurId",
  "modification": true,
  "nom": "utilisateurNom",
  "permissionModification": true,
  "preferences": null,
  "prenom": null,
  "qgisToken": null,
  "role": "editeur",
  "suppression": true,
  "telephoneFixe": null,
  "telephoneMobile": null,
}
`;

exports[`utilisateursQueryModify Vérifie l'écriture de la requête sur un utilisateur 2`] = `
Object {
  "administration": Object {
    "activitesTypes": Array [],
    "id": "aut-97300-01",
    "titresTypes": Array [
      Object {
        "archive": null,
        "associee": true,
        "contenuIds": Array [
          Object {
            "elementId": "mecanise",
            "sectionId": "arm",
          },
          Object {
            "elementId": "franchissements",
            "sectionId": "arm",
          },
        ],
        "domaineId": "m",
        "gestionnaire": null,
        "id": "arm",
        "typeId": "ar",
      },
    ],
  },
  "administrationId": "aut-97300-01",
  "dateCreation": "2022-05-12",
  "email": "utilisateurEmail",
  "entreprises": Array [],
  "id": "utilisateurId",
  "modification": true,
  "nom": "utilisateurNom",
  "permissionModification": true,
  "preferences": null,
  "prenom": null,
  "qgisToken": null,
  "role": "editeur",
  "suppression": true,
  "telephoneFixe": null,
  "telephoneMobile": null,
}
`;

exports[`utilisateursQueryModify Vérifie l'écriture de la requête sur un utilisateur 3`] = `
Object {
  "administration": Object {
    "activitesTypes": Array [],
    "id": "aut-97300-01",
    "titresTypes": Array [
      Object {
        "archive": null,
        "associee": true,
        "contenuIds": Array [
          Object {
            "elementId": "mecanise",
            "sectionId": "arm",
          },
          Object {
            "elementId": "franchissements",
            "sectionId": "arm",
          },
        ],
        "domaineId": "m",
        "gestionnaire": null,
        "id": "arm",
        "typeId": "ar",
      },
    ],
  },
  "administrationId": "aut-97300-01",
  "dateCreation": "2022-05-12",
  "email": "utilisateurEmail",
  "entreprises": Array [],
  "id": "utilisateurId",
  "modification": false,
  "nom": "utilisateurNom",
  "permissionModification": false,
  "preferences": null,
  "prenom": null,
  "qgisToken": null,
  "role": "editeur",
  "suppression": false,
  "telephoneFixe": null,
  "telephoneMobile": null,
}
`;

exports[`utilisateursQueryModify Vérifie l'écriture de la requête sur un utilisateur 4`] = `
Object {
  "administration": Object {
    "activitesTypes": Array [],
    "id": "aut-97300-01",
    "titresTypes": Array [
      Object {
        "archive": null,
        "associee": true,
        "contenuIds": Array [
          Object {
            "elementId": "mecanise",
            "sectionId": "arm",
          },
          Object {
            "elementId": "franchissements",
            "sectionId": "arm",
          },
        ],
        "domaineId": "m",
        "gestionnaire": null,
        "id": "arm",
        "typeId": "ar",
      },
    ],
  },
  "administrationId": "aut-97300-01",
  "dateCreation": "2022-05-12",
  "email": "utilisateurEmail",
  "entreprises": Array [],
  "id": "utilisateurId",
  "modification": false,
  "nom": "utilisateurNom",
  "permissionModification": false,
  "preferences": null,
  "prenom": null,
  "qgisToken": null,
  "role": "editeur",
  "suppression": false,
  "telephoneFixe": null,
  "telephoneMobile": null,
}
`;

exports[`utilisateursQueryModify Vérifie l'écriture de la requête sur un utilisateur 5`] = `
Object {
  "administration": Object {
    "activitesTypes": Array [],
    "id": "aut-97300-01",
    "titresTypes": Array [
      Object {
        "archive": null,
        "associee": true,
        "contenuIds": Array [
          Object {
            "elementId": "mecanise",
            "sectionId": "arm",
          },
          Object {
            "elementId": "franchissements",
            "sectionId": "arm",
          },
        ],
        "domaineId": "m",
        "gestionnaire": null,
        "id": "arm",
        "typeId": "ar",
      },
    ],
  },
  "administrationId": "aut-97300-01",
  "dateCreation": "2022-05-12",
  "email": "utilisateurEmail",
  "entreprises": Array [],
  "id": "utilisateurId",
  "modification": false,
  "nom": "utilisateurNom",
  "permissionModification": false,
  "preferences": null,
  "prenom": null,
  "qgisToken": null,
  "role": "editeur",
  "suppression": false,
  "telephoneFixe": null,
  "telephoneMobile": null,
}
`;
